assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -269,9 +271,11 @@ export default function CurrentSurvey() {
 				<SurveyQuestionComponent
 					key={ currentQuestion.question_text }
 					answerQuestion={ answerQuestion }
-					choices={ currentQuestion.question.answer_choice }
+					choices={ currentQuestion.question?.answer_choice } // eslint-disable-line camelcase
```

Comment:
Why the optional chaining here?

Reply:
This type of question has no choices, so this was the least merge-conflict way of solving this :smile: 

Now this is the last PR to be merged, we could make changes here.

IMHO we should get rid of the `ComponentMap` and just do a switch statement, returning the correct component with the correct props

Reply:
Ahhh, okay, right, makes sense.

Yeah, I think I agree. There's no sense in this approach, and it's less obvious.

Good call on waiting until the last PR though 🙂

I'm fine with a switch case, but if it's not a giant pain/unreadable to do `currentQuestion?.question_type === 'single_select' && <SurveyQuestionSingleSelect />` I'd slightly prefer it. Since we can't put a switch case in the JSX is all, and I find it a bit easier to read about the component output the more we can put all in the JSX that's returned.

But if it makes it really bad to read we can do the switch case instead.

Agreed, let's do that 👍🏻

Reply:
@tofumatt  Apologies but I'm not following the sentence `if it's not a giant pain/unreadable to do`

In the comment at the end of this PR you mention to just do the switch statement so I'll do that now, make sure the tests pass, and then we could sync to discuss your other idea further :smile:

Reply:
I made the change and the tests pass for me locally. Will assign to you when all the GitHub actions pass :+1: